### PR TITLE
populate_game! method in game.rb

### PR DESCRIPTION
**Pull request by bhk1019**
Hi everyone, here's a draft for the populate_game! method in game.rb.

Main additions:

- wrote a populate_game! method that generates 32 new Piece objects, and links their game_id to the parent game that called the method.
- added after_create method to populate_game! so that this method is called whenever a new game is created.
- wrote two examples inside game_spec.rb that successfully pass. One test checks that 32 pieces were created that belong_to the game, and another test checks that the two generated queens are in the right positions.

Other notes:

- I had to add `optional: true` to the `white_player` and `black_player` associations inside game.rb in order for the tests to work. It doesn't seem like there will be any merge conflicts, however.
- I deleted the pending tests inside the other spec files for each of the individual piece classes.
- I had to run migrations to update the schema but again, I don't think this conflicts with anything.

### CODE REVIEW

**Comment by bhk1019** [comment]
Ok an update: I had to add redundant migration files because travis was building the entire app from scratch in order to run the tests. On my machine, it was not accepting the migrations (attribute already exists error) because somehow the schema already had the player ids and game id columns even without the migration files. But looking at it now, it passes the travis checks with migration files added!!

File: app/models/game.rb [hunk 1]
```diff
@@ -1,5 +1,70 @@
...
+
+  def populate_game!
+    [true, false].each do |is_white|
+      y = is_white ? 1 : 6
+      8.times do |x|
+        Pawn.create(
+          :game_id => self.id, 
+          :x_coord => x, 
+          :y_coord => y, 
+          :type => 'Pawn',
```
Comment (ilyakrasnov) [inline app/models/game.rb]: No need for type, since you are running the `Pawn.create(..`

Comment (ilyakrasnov) [reply app/models/game.rb]: I'd suggest to bring the whole `create..` to one line - just for the sake of readability and to make the code more scannable...

**Comment by bhk1019** [comment]
Made changes as per @ilyakrasnov 's suggestions and fixed conflicts.

**Pull request merged**